tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7473,6 +7473,14 @@ def err_hlsl_attribute_mismatch: Error<
   "invalid shader stage attribute combination">;
 def err_hlsl_attribute_valid_on_function_only: Error<
   "attribute is valid only on functions">;
+def err_hlsl_attribute_valid_on_entry_function_only: Error<
+  "%0 attribute only valid on entry point functions">;
+def err_hlsl_attr_for_wrong_shader: Error<
+  "attribute %0 only valid for %1.">;
+def err_hlsl_numthreads_attr: Error<
+  "attribute numthreads only valid for CS/MS/AS.">;
```

Comment:
This error could just reuse the error above with a constructed string. It is also probably better to put the full stage names in a sentence like structure: "compute, mesh, or amplification".

Reply:
Excellent simplification, fixed.